cmd/utils/flags.go

```diff
@@ -1449,6 +1452,22 @@ func RegisterEthService(stack *node.Node, cfg *eth.Config) {
 		err = stack.Register(func(ctx *node.ServiceContext) (node.Service, error) {
 			return les.New(ctx, cfg)
 		})
+		if err == nil {
+			err = stack.RegisterCallback(reflect.TypeOf(&les.LightEthereum{}), func(service node.Service) error {
```

Comment:
Having an extra startup callback mechanism in node.Node seems redundant since each service has its Start function already. Also we have this ugly issue with AttachLocked which is definitely not good. Can't you pass a callback function to les.New and eth.New during service registration that attaches a new client and returns it? Then LightEthereum.Start could call this when the server is already running and set the contract backend itself (SetContractBackend is also not needed then).

Reply:
Unfortunately, we start all services first(invoke services' `Start` function) and then initialize all RPC services. 
Attaching local client requires all RPC services have been opened. So this approach doesn't work.